pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -0,0 +1,258 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"math/big"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/vmware/go-ipfix/pkg/registry"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/flowexporter"
+	"antrea.io/antrea/pkg/agent/flowexporter/connections"
+	"antrea.io/antrea/pkg/agent/flowexporter/flowrecords"
+)
+
+const (
+	testNumOfConns           = 20000
+	testNumOfDenyConns       = 20000
+	testNumOfDyingConns      = 2000
+	testNumOfInactiveRecords = 2000
+	testNumOfIdleDenyConns   = 2000
+	testBufferSize           = 1048
+)
+
+var recordsReceived = 0
+
+/*
+Sample output:
+	go test -test.v -run=BenchmarkExport -test.benchmem -bench=BenchmarkExportConntrackConns -memprofile memprofile.out -cpuprofile profile.out
+	goos: linux
+	goarch: amd64
+	pkg: antrea.io/antrea/pkg/agent/flowexporter/exporter
+	cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
+	BenchmarkExportConntrackConns (truncated output)
+		exporter_perf_test.go:79:
+			Summary:
+			Number of conntrack connections: 20000
+			Number of dying conntrack connections: 2000
+			Total connections received: 18703
+	BenchmarkExportConntrackConns-2   	      75	  13750074 ns/op	  965550 B/op	   22268 allocs/op
+	PASS
+	ok  	antrea.io/antrea/pkg/agent/flowexporter/exporter	5.494s
+*/
+func BenchmarkExportConntrackConns(b *testing.B) {
+	disableLogToStderr()
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+
+	recordsReceived = 0
+	exp, err := setupExporter(true)
+	if err != nil {
+		b.Fatalf("error when setting up exporter: %v", err)
+	}
+	for n := 0; n < b.N; n++ {
+		exp.initFlowExporter()
+		exp.sendFlowRecords()
+	}
+	b.Logf("\nSummary:\nNumber of conntrack connections: %d\nNumber of dying conntrack connections: %d\nTotal connections received: %d\n", testNumOfConns, testNumOfDyingConns, recordsReceived)
+}
+
+/*
+Sample output:
+	go test -test.v -run=BenchmarkExport -test.benchmem -bench=BenchmarkExportDenyConns -memprofile memprofile.out -cpuprofile profile.out
+	goos: linux
+	goarch: amd64
+	pkg: antrea.io/antrea/pkg/agent/flowexporter/exporter
+	cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
+	BenchmarkExportDenyConns (truncated output)
+		exporter_perf_test.go:112:
+			Summary:
+			Number of deny connections: 20000
+			Number of idle deny connections: 2000
+			Total connections received: 19124
+	BenchmarkExportDenyConns-2   	     204	   6922004 ns/op	  357215 B/op	   12106 allocs/op
+	PASS
+	ok  	antrea.io/antrea/pkg/agent/flowexporter/exporter	6.189s
+*/
+func BenchmarkExportDenyConns(b *testing.B) {
+	disableLogToStderr()
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+
+	recordsReceived = 0
+	exp, err := setupExporter(false)
+	if err != nil {
+		b.Fatalf("error when setting up exporter: %v", err)
+	}
```

Comment:
The golang testing documentation recommends resetting the timer after doing an "expensive" setup: https://golang.org/pkg/testing/#hdr-Benchmarks

Can you call `b.ResetTimer()` in all you benchmarks, just before the loop?

Reply:
Sure. Thanks!